rasterio/dtypes.py

```diff
@@ -77,12 +77,11 @@ def _gdal_typename(dt):
 
 def check_dtype(dt):
     """Check if dtype is a known dtype."""
-    if dt not in dtype_rev:
-        try:
-            return dt().dtype.name in dtype_rev
-        except:
-            return False
-    return True
+    if str(dt) in dtype_rev:
+        return True
+    elif callable(dt) and str(dt().dtype) in dtype_rev:
+        return True
+    return False
```

Comment:
per chat, this is what is causing the current master to fail on travis

 https://travis-ci.org/mapbox/rasterio/jobs/242292775